File: backend/src/Tmap.Api/Features/Auth/AuthDtos.cs
namespace Tmap.Api.Features.Auth;

public sealed record RegisterRequest(string Email, string Password);
public sealed record LoginRequest(string Email, string Password);
// Native clients send the refresh token in the body; web omits it (cookie). Nullable for that reason.
public sealed record RefreshRequest(string? RefreshToken);
public sealed record LogoutRequest(string? RefreshToken);

/// <summary>Public OpenAPI-visible response for register / login / refresh.</summary>
/// <param name="AccessToken">The access token.</param>
/// <param name="RefreshToken">Refresh token — absent on web (httpOnly cookie); present for native clients.</param>
/// <param name="ExpiresIn">Seconds until the access token expires.</param>
/// <param name="User">The authenticated user's public profile.</param>
public sealed record AuthTokenResponse(
    string AccessToken,
    string RefreshToken,
    int ExpiresIn,
    AuthTokenUser User);

public sealed record AuthTokenUser(Guid Id, string Email, string TimeZoneId);

// Internal: full issuer result; converted to AuthTokenResponse before returning.
internal sealed record TokenPairResponse(string AccessToken, string RefreshToken, DateTimeOffset AccessTokenExpiresAt);
public sealed record MeResponse(Guid Id, string Email, string TimeZoneId);

File: android/app/src/main/java/net/qmindtech/tmap/data/remote/dto/AuthDtos.kt
package net.qmindtech.tmap.data.remote.dto

import kotlinx.serialization.Serializable

@Serializable
data class RegisterRequest(
    val email: String,
    val password: String,
)

@Serializable
data class LoginRequest(
    val email: String,
    val password: String,
)

@Serializable
data class RefreshRequest(
    val refreshToken: String,
)

@Serializable
data class LogoutRequest(
    val refreshToken: String,
)

@Serializable
data class AuthTokenUser(
    val id: String,
    val email: String,
    val timeZoneId: String,
)

@Serializable
data class AuthTokenResponse(
    val accessToken: String,
    val refreshToken: String? = null,
    val expiresIn: Int,
    val user: AuthTokenUser,
)
